crates/nargo_cli/src/cli/prove_cmd.rs

```diff
@@ -78,20 +78,15 @@ pub(crate) fn prove_with_path<P: AsRef<Path>>(
 
     let PreprocessedProgram { abi, bytecode, proving_key, verification_key, .. } =
         preprocessed_program;
-    let compiled_program = CompiledProgram { abi, circuit: bytecode };
 
     // Parse the initial witness values from Prover.toml
-    let (inputs_map, _) = read_inputs_from_file(
-        &program_dir,
-        PROVER_INPUT_FILE,
-        Format::Toml,
-        &compiled_program.abi,
-    )?;
+    let (inputs_map, _) =
+        read_inputs_from_file(&program_dir, PROVER_INPUT_FILE, Format::Toml, &abi)?;
 
-    let solved_witness = execute_program(&backend, &compiled_program, &inputs_map)?;
+    let solved_witness = execute_program(&backend, bytecode.clone(), &abi, &inputs_map)?;
```

Comment:
Note: This clone was there before, and is an artifact of the partial witness generator taking ownership (perhaps in the future, we can make it use a mutable borrow) not important right now.